reframe/core/modules.py

```diff
@@ -410,25 +410,25 @@ def name(self):
     def version(self):
         return self._version
 
-    def _run_module_command(self, *args):
-        command = [self._command, *args]
-        return os_ext.run_command(' '.join(command))
+    def _run_module_command(self, *args, msg=None):
+        command = ' '.join([self._command, *args])
+        try:
+            completed = os_ext.run_command(command, check=True)
+        except SpawnedProcessError as e:
+            raise EnvironError(msg) from e
+
+        if self._module_command_failed(completed):
+            raise EnvironError(msg) from SpawnedProcessError(
+                command, completed.stdout, completed.stderr,
+                completed.returncode)
```

Comment:
It would be better to format this as follows:

```python
err = SpawnedProcessError(command,
                          completed.stdout,
                          completed.stderr,
                          completed.returncode)
raise EnvironError(msg) from err
```

Reply:
Done.